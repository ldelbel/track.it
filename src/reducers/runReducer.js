import { ADD_RUNNING_SESSION, SET_USER, FILL_LIST } from '../actions';

let initialState = {
  id: 0,
  username: '',
  runningSessions: [],
};

if (localStorage.user) {
  initialState = {
    id: localStorage.id,
    username: localStorage.user,
    runningSessions: JSON.parse(localStorage.runningSessions),
  };
}

const runReducer = (state = initialState, action) => {
  switch (action.type) {
    case ADD_RUNNING_SESSION:
      return {
        ...state,
        runningSessions: state.runningSessions.concat(action.payload),
      };
    case SET_USER:
      return {
        ...state,
        username: action.payload,
      };
    case FILL_LIST:
      return {
        ...state,
        id: action.payload[0],
        runningSessions: action.payload[1],
      };
    default:
      return state;
  }
};

export default runReducer;
