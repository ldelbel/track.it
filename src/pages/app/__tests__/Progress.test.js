import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import Progress from '../progress';
import runReducer from '../../../reducers/runReducer';


jest.mock('../progress/Chart', () => 'Chart')

describe('progress component',() => {
    let running_sessions = [
        {
            id: 1,
            user: 'test1',
            distance: 1,
            duration: 1,
            start_time: 1616593525386,
            avg_speed: 1,
            avg_pace: 1,
            goal: 1,
        },{
            id: 2,
            user: 'test2',
            distance: 2,
            duration: 2,
            start_time: 1616593525386,
            avg_speed: 2,
            avg_pace: 2,
            goal: 2,
        }
    ]

    it('renders correctly', () => {
        const store = createStore(runReducer);
        const progress = renderer.create(
          <Provider store={store}>
            <Progress runningSessions={running_sessions} />
          </Provider>,
        ).toJSON();
        expect(progress).toMatchSnapshot();
    });
})