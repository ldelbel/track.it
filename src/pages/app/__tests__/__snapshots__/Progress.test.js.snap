// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`progress component renders correctly 1`] = `
<main
  className="container"
>
  <div
    className="content"
  >
    <div
      className="content__title"
    >
      Your Progress
    </div>
    <div
      className="content__chart"
    >
      <Chart
        distancesPerDay={
          Array [
            Array [
              "Mar 24",
              1,
            ],
            Array [
              "Mar 24",
              2,
            ],
          ]
        }
      />
    </div>
    <div
      className="section"
    >
      <div
        className="section__title"
      >
        Overall
      </div>
      <div
        className="section__info"
      >
        <div
          className="section__info__item"
        >
          <span>
            Avg Speed
          </span>
          <div
            className="section__info__item__div"
            div={true}
          >
            <div>
              <span>
                1.5
              </span>
              <p>
                km/h
              </p>
            </div>
          </div>
        </div>
        <div
          className="section__info__item"
        >
          <span>
            Avg Pace
          </span>
          <div
            className="section__info__item__div"
            div={true}
          >
            <div>
              <span>
                1.5
              </span>
              <p>
                h/km
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="section"
    >
      <div
        className="section__title"
      >
        Personal Best
      </div>
      <div
        className="section__info"
      >
        <div
          className="section__info__item"
        >
          <span>
            Speed
          </span>
          <div
            className="section__info__item__div"
            div={true}
          >
            <div>
              <span>
                2
              </span>
              <p>
                km/h
              </p>
            </div>
          </div>
        </div>
        <div
          className="section__info__item"
        >
          <span>
            Pace
          </span>
          <div
            className="section__info__item__div"
            div={true}
          >
            <div>
              <span>
                2
              </span>
              <p>
                h/km
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</main>
`;
